src/app/courses/courses.component.ts

```diff
@@ -322,7 +322,7 @@ export class CoursesComponent implements OnInit, AfterViewInit, OnDestroy {
       const resourcesToSend = [].concat.apply([], coursesToSend.map(course =>
           [].concat.apply([], course.steps.map(step => step.resources))
         ));
-      const examsToSend = [].concat.apply([], coursesToSend.map(course => course.steps.map(step => step.exam)));
+      const examsToSend = [].concat.apply([], coursesToSend.map(course => [].concat.apply([], course.steps.map(step => step.exam || []))));
```

Comment:
We should refactor to avoid so much nesting.

Reply:
updated now to use flatmap